### ISSUE TITLE
Consider making different naming rules for variables and types

### ISSUE THREAD

**Issue by hkim15** [project-oak/hafnium-verification#24]
In the current Hafnium code, it is difficult to tell if an identifier is a type or a variable. It might be useful to create naming rules to differentiate them.

**Comment by jeehoonkang** [comment]
I disagree with you on the idea of such naming convention.  The primary reason is that types and variables are easily distinguishable by modern IDEs.  That's also one of the reasons why the ["hungarian notation"](https://en.wikipedia.org/wiki/Hungarian_notation) is unused in most modern development projects.

**Comment by hkim15** [comment]
I am also oppose to something like "hungarian notation". On the other hand, I am in more favor of using one of the following variations to distinguish different classes of identifiers:

ALL_UPPERCASES
all_lowercases
MixedCasesThatBeginWithUppercase
mixedCasesThatBeginWithLowercase

For instance, I favor using ALL_UPPERCASES for constants and MixedCasesThatBeginWithUppercase for types, member functions/variables. For local variables, I prefer using all_lowercases or mixedCasesThatBeginsWithLowercase.
